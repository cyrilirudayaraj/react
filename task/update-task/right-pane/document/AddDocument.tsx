import React from 'react';
import { Button, Lightbox, Form, FormField, Select } from '@athena/forge';
import { ButtonVariant } from '@athena/forge/Button/Button';
import { Formik, FormikProps } from 'formik';
import * as Yup from 'yup';
import { AttachmentDetail } from '../../../../../types';
import Labels from '../../../../../constants/Labels';

interface AddDocumenttProps {
  onConfirm: (values: any) => any;
  className: string;
  id: string;
  headerText: string;
  context: string;
  icon: string;
  variant: ButtonVariant;
  disabled: boolean;
  text?: string;
  attachment?: AttachmentDetail;
  type?: string[];
}
type ModalState = {
  shown: boolean;
};

class AddDocument extends React.Component<AddDocumenttProps> {
  state = {
    shown: false,
  };

  showLightbox = (formik: FormikProps<any>) => {
    this.setState({
      shown: true,
    });
    if (this.props.context === Labels.DOCUMENTS.CONTEXT_EDIT) {
      formik.setValues(this.props.attachment);
    } else if (this.props.context === Labels.DOCUMENTS.CONTEXT_ADD) {
      formik.resetForm();
    }
  };
  hideLightbox = (): void => {
    this.setState({
      shown: false,
    });
  };

  handleAttachmentTypeIdChange = (event: any, formik: any): void => {
    formik.setFieldValue(event.target.id, event.target.value);
  };

  initialValues = {
    fileName: this.props.attachment?.fileName || '',
    filePath: this.props.attachment?.filePath || '',
    attachmentTypeId: this.props.attachment?.attachmentTypeId || '',
    description: this.props.attachment?.description || '',
  };

  validationSchema = function () {
    return Yup.object().shape({
      filePath: Yup.string().required('Required!'),
      fileName: Yup.string().required('Required!'),
      attachmentTypeId: Yup.string().required('Required!'),
    });
  };

  getFormFieldProps = (fieldName: string, formik: any) => {
    const props = {
      ...formik.getFieldProps(fieldName),
      error: formik.errors[fieldName],
      id: fieldName,
      labelWidth: 12,
      required: true,
    };
    return props;
  };

  renderForm = (formik: FormikProps<any>) => {
    return (
      <Form
        labelAlwaysAbove={true}
        includeSubmitButton={false}
        requiredVariation="blueBarWithLegend"
        className="fe_u_margin--large document-form"
        layout="medium"
      >
        <FormField
          maxlength="1000"
          labelText="Link"
          {...this.getFormFieldProps('filePath', formik)}
        />
        <FormField
          maxlength="200"
          labelText="Name"
          {...this.getFormFieldProps('fileName', formik)}
        />
        <FormField
          labelText="Type"
          inputAs={Select}
          options={this.props.type}
          {...this.getFormFieldProps('attachmentTypeId', formik)}
          onChange={(event: any) => {
            this.handleAttachmentTypeIdChange(event, formik);
          }}
        />
        <FormField
          labelText="Notes"
          {...this.getFormFieldProps('description', formik)}
          maxLength="1000"
          required={false}
        />
      </Form>
    );
  };
  render(): JSX.Element {
    return (
      <div className={this.props.className}>
        <Formik
          initialValues={this.initialValues}
          validationSchema={this.validationSchema}
          onSubmit={(values) => {
            this.props.onConfirm({
              ...values,
              id: this.props?.attachment?.id,
              version: this.props?.attachment?.version,
            });
            this.hideLightbox();
          }}
          validateOnChange={false}
          validateOnBlur={false}
          validateOnMount={false}
        >
          {(formik) => {
            return (
              <>
                <Button
                  className={this.props.className}
                  id={this.props.id}
                  text={this.props.text}
                  icon={this.props.icon}
                  variant={this.props.variant}
                  disabled={this.props.disabled}
                  onClick={() => this.showLightbox(formik)}
                />
                <Lightbox
                  show={this.state.shown}
                  hideDividers
                  headerText={this.props.headerText}
                  disableClose
                  width="large"
                  className="my-custom-lightbox create_document"
                >
                  <Button
                    variant="tertiary"
                    icon={Labels.DOCUMENTS.ICON_CLOSE}
                    onClick={() => this.hideLightbox()}
                    className="my-close-button"
                  />
                  {this.renderForm(formik)}
                  <div className="fe_c_lightbox__footer">
                    <Button
                      text={Labels.DOCUMENTS.BUTTON_CANCEL}
                      variant="secondary"
                      className="fe_u_margin--right-small"
                      onClick={() => this.hideLightbox()}
                    />
                    <Button
                      text={this.props.context}
                      onClick={(values) => {
                        formik.handleSubmit(values);
                      }}
                    />
                  </div>
                </Lightbox>
              </>
            );
          }}
        </Formik>
      </div>
    );
  }
}

export default AddDocument;
