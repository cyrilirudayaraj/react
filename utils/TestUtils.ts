//TODO exclude this file during bundling
export function findByName(wrapper: any, name: string): any {
  const selector = "input[name='" + name + "']";
  return wrapper.find(selector).last();
}

export function findById(wrapper: any, id: string): any {
  const selector = '#' + id;
  return wrapper.find(selector).last();
}

export function changeValueById(wrapper: any, id: string, value: string): void {
  findById(wrapper, id).simulate('change', {
    target: { name: id, value: value },
  });
}

export function sleep(x: number): Promise<any> {
  return new Promise<void>((resolve) => {
    setTimeout(() => {
      resolve();
    }, x);
  });
}
