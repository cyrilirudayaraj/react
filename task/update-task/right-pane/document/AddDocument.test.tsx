import React from 'react';
import { configure, mount } from 'enzyme';
import AddDocument from './AddDocument';

import Adapter from 'enzyme-adapter-react-16';
import { AttachmentDetail } from '../../../../../types';
import { act } from 'react-dom/test-utils';
import Labels from '../../../../../constants/Labels';
import { FormikProps } from 'formik';
configure({ adapter: new Adapter() });
jest.mock('../../../../../services/CommonService');
import { findByName } from '../../../../../utils/TestUtils';

const attachment: AttachmentDetail = {
  id: '1',
  taskId: '1',
  taskStepId: '16',
  description: 'Test cyril content 1',
  fileName: 'Marksheet',
  filePath: 'https://www.drive.in/cover',
  attachmentName: 'Local Rule Authorization',
  attachmentTypeId: '1',
  createdBy: 'jcyril',
  version: '1',
};

const attachmentType: any = [
  { value: '1', text: 'Local Rule Authorization' },
  { value: '2', text: 'Payer Documentation' },
  { value: '3', text: 'Data Support' },
  { value: '4', text: 'Athenanet Screenshot' },
  { value: '5', text: 'Client clarification' },
];

let wrapper: any;
let updateAttachment: any;
let formikProps: FormikProps<any>;
let documentComponent: any;
beforeEach(() => {
  updateAttachment = jest.fn();
  formikProps = {} as FormikProps<any>;
  formikProps = {
    ...formikProps,
    values: '',
    touched: {},
    errors: {},
    handleChange: (a: any) => a,
    handleBlur: (a: any) => a,
    handleSubmit: (a) => a,
    setValues: (a: any) => a,
    resetForm: jest.fn(),
    getFieldProps: jest.fn(),
    setFieldValue: jest.fn(),
  };
  wrapper = mount(
    <AddDocument
      className="edit-button"
      id="edit-button"
      headerText={Labels.DOCUMENTS.HEADER_TEXT_EDIT}
      context={Labels.DOCUMENTS.CONTEXT_EDIT}
      icon={Labels.DOCUMENTS.ICON_EDIT}
      variant="tertiary"
      onConfirm={updateAttachment}
      attachment={attachment}
      type={attachmentType}
    />
  );
  wrapper.setState({ shown: true });
  documentComponent = mount(
    <AddDocument
      className="add-button"
      id="add-button"
      headerText={Labels.DOCUMENTS.HEADER_TEXT_ADD}
      context={Labels.DOCUMENTS.CONTEXT_ADD}
      icon={Labels.DOCUMENTS.ICON_ADD}
      variant="tertiary"
      onConfirm={updateAttachment}
      attachment={attachment}
      type={attachmentType}
    />
  );
});

describe('AddDocument', () => {
  it('basic snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should close the dialog when close button clicked', () => {
    act(() => {
      wrapper.find('.my-close-button').last().simulate('click');
      wrapper.instance().hideLightBox;
      expect(wrapper.state().shown).toEqual(true);
    });
  });
  it('should show the edit dialog when edit button clicked', () => {
    act(() => {
      wrapper.find('button#edit-button').last().simulate('click');
      wrapper.instance().showLightbox(formikProps);
      expect(wrapper.state().shown).toEqual(true);
    });
  });
  it('should show the add dialog when add button clicked', () => {
    act(() => {
      documentComponent.find('button#add-button').last().simulate('click');
      documentComponent.instance().showLightbox(formikProps);
      expect(wrapper.state().shown).toEqual(true);
    });
  });

  it('should close the dialog when cancel button clicked', () => {
    act(() => {
      wrapper.find('.fe_c_button--secondary').last().simulate('click');
      wrapper.instance().hideLightBox;
      expect(wrapper.state().shown).toEqual(true);
    });
  });

  it('should fill the form and submit', async () => {
    await act(async () => {
      setAttachmentType();
      findByName(wrapper, 'fileName').simulate('change', {
        target: { name: 'fileName', value: 'CPC codes' },
      });
      findByName(wrapper, 'filePath').simulate('change', {
        target: { name: 'filePath', value: 'https://www.google.drive.in' },
      });
      findByName(wrapper, 'description').simulate('change', {
        target: { name: 'description', value: 'test CPC codes' },
      });
      wrapper.find('.fe_c_button--primary').simulate('click');
    });
  });
});

function setAttachmentType(): void {
  wrapper
    .find('#attachmentTypeId')
    .last()
    .simulate('change', {
      target: { name: 'attachmentTypeId', value: '3', selectedIndex: 1 },
    });
}
