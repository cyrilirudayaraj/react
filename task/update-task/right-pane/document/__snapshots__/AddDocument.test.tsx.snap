// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDocument basic snapshot 1`] = `
"<AddDocument className=\\"edit-button\\" id=\\"edit-button\\" headerText=\\"Edit Document\\" context=\\"Save\\" icon=\\"Edit\\" variant=\\"tertiary\\" onConfirm={[Function: mockConstructor]} attachment={{...}} type={{...}}>
  <div className=\\"edit-button\\">
    <Formik initialValues={{...}} validationSchema={[Function (anonymous)]} onSubmit={[Function: onSubmit]} validateOnChange={false} validateOnBlur={false} validateOnMount={false}>
      <ForwardRef className=\\"edit-button\\" id=\\"edit-button\\" text={[undefined]} icon=\\"Edit\\" variant=\\"tertiary\\" disabled={false} onClick={[Function: onClick]} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
        <button id=\\"edit-button\\" className=\\"fe_c_button fe_c_button--tertiary fe_c_button--medium edit-button\\" aria-disabled={false} aria-label=\\"Edit\\" onClick={[Function: handleClick]} type=\\"button\\">
          <div className=\\"fe_c_button__content\\">
            <Edit title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--tertiary\\">
              <IconBase viewBox=\\"0 0 25 25\\" title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--tertiary\\">
                <svg xmlns=\\"http://www.w3.org/2000/svg\\" className=\\"fe_c_icon fe_c_button__icon fe_c_button__icon--tertiary\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                  <g>
                    <g className=\\"fe_c_icon__background\\">
                      <rect x=\\"-0.2\\" y=\\"-0.2\\" width=\\"25\\" height=\\"25\\" />
                    </g>
                    <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                      <path d=\\"M21.1,4.8,16.9.7,4.1,13.4,2.6,21.5l8-1.6L21.1,9.5A3.3,3.3,0,0,0,21.1,4.8ZM5.2,15.6l3.3,3.2-4,.8Zm4.9,2.8L5.7,13.9l8-8,4.5,4.4ZM20,8.5l-.8.8L14.8,4.9l2.1-2.1L20,5.9A1.9,1.9,0,0,1,20,8.5Z\\" />
                    </g>
                  </g>
                </svg>
              </IconBase>
            </Edit>
          </div>
        </button>
      </ForwardRef>
      <ThemeConsumerHOC show={true} hideDividers={true} headerText=\\"Edit Document\\" disableClose={true} width=\\"large\\" className=\\"my-custom-lightbox create_document\\">
        <Lightbox show={true} hideDividers={true} headerText=\\"Edit Document\\" disableClose={true} width=\\"large\\" className=\\"my-custom-lightbox create_document\\" theme={{...}} closeButtonAriaLabel=\\"Close Modal\\" includeButtonRow={false} primaryButtonText=\\"Next\\" padded={true}>
          <ModalManagerWrapper className=\\"fe_c_lightbox fe_f_all my-custom-lightbox create_document\\" dialogClass={{...}} onHide={[Function: handleOnHide]} aria-label=\\"Edit Document\\" show={true}>
            <Overlay manager={{...}} className=\\"fe_c_lightbox fe_f_all my-custom-lightbox create_document\\" dialogClass={{...}} onHide={[Function: handleOnHide]} aria-label=\\"Edit Document\\" show={true} autoFocus={false} enforceFocus={false}>
              <Modal aria-modal=\\"true\\" aria-labelledby={[undefined]} aria-describedby={[undefined]} renderBackdrop={[Function: renderBackdrop]} transition={[Function: FadeUp]} backdropTransition={[Function: Fade]} manager={{...}} onShow={[Function (anonymous)]} onHide={[Function: handleOnHide]} aria-label=\\"Edit Document\\" show={true} autoFocus={false} enforceFocus={false} className=\\"fe_c_overlay fe_f_all fe_c_lightbox fe_f_all my-custom-lightbox create_document\\" role=\\"dialog\\" backdrop={true} keyboard={true} restoreFocus={true}>
                <Portal container={[undefined]} onRendered={[Function (anonymous)]}>
                  <WaitForContainer container={[undefined]} onContainerResolved={[Function (anonymous)]}>
                    <Portal containerInfo={{...}}>
                      <Fade appear={true} in={true}>
                        <CSSTransition timeout={400} classNames=\\"fade\\" appear={true} in={true}>
                          <Transition timeout={400} appear={true} in={true} onEnter={[Function (anonymous)]} onEntered={[Function (anonymous)]} onEntering={[Function (anonymous)]} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} onExited={[Function (anonymous)]} mountOnEnter={false} unmountOnExit={false} enter={true} exit={true}>
                            <div className=\\"fe_c_overlay__blocker \\" onClick={[Function (anonymous)]} />
                          </Transition>
                        </CSSTransition>
                      </Fade>
                      <FadeUp appear={true} unmountOnExit={true} in={true} onExit={[undefined]} onExiting={[undefined]} onExited={[Function (anonymous)]} onEnter={[undefined]} onEntering={[undefined]} onEntered={[undefined]}>
                        <CSSTransition timeout={400} classNames=\\"fade-up\\" appear={true} unmountOnExit={true} in={true} onExit={[undefined]} onExiting={[undefined]} onExited={[Function (anonymous)]} onEnter={[undefined]} onEntering={[undefined]} onEntered={[undefined]}>
                          <Transition timeout={400} appear={true} unmountOnExit={true} in={true} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} onExited={[Function (anonymous)]} onEnter={[Function (anonymous)]} onEntering={[Function (anonymous)]} onEntered={[Function (anonymous)]} mountOnEnter={false} enter={true} exit={true}>
                            <div role=\\"dialog\\" aria-modal=\\"true\\" aria-labelledby={[undefined]} aria-describedby={[undefined]} aria-label=\\"Edit Document\\" style={[undefined]} className=\\"fe_c_overlay fe_f_all fe_c_lightbox fe_f_all my-custom-lightbox create_document\\" tabIndex=\\"-1\\">
                              <div className=\\"fe_c_overlay__dialog fe_c_lightbox__dialog fe_c_lightbox__dialog--large fe_c_lightbox__dialog--without-dividers\\" role=\\"document\\">
                                <div className=\\"fe_c_lightbox__header\\">
                                  <Heading text=\\"Edit Document\\" headingTag=\\"h1\\" variant=\\"section\\">
                                    <h1 className=\\"fe_c_heading fe_c_heading--section\\">
                                      Edit Document
                                    </h1>
                                  </Heading>
                                </div>
                                <div className=\\"fe_c_lightbox__body fe_c_lightbox__body--padded\\">
                                  <ForwardRef variant=\\"tertiary\\" icon=\\"Close\\" onClick={[Function: onClick]} className=\\"my-close-button\\" disabled={false} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
                                    <button className=\\"fe_c_button fe_c_button--tertiary fe_c_button--medium my-close-button\\" aria-disabled={false} aria-label=\\"Close\\" onClick={[Function: handleClick]} type=\\"button\\">
                                      <div className=\\"fe_c_button__content\\">
                                        <Close title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--tertiary\\">
                                          <IconBase viewBox=\\"0 0 25 25\\" title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--tertiary fe_c_icon--close\\">
                                            <svg xmlns=\\"http://www.w3.org/2000/svg\\" className=\\"fe_c_icon fe_c_button__icon fe_c_button__icon--tertiary fe_c_icon--close\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                                              <g>
                                                <g className=\\"fe_c_icon__background\\">
                                                  <rect width=\\"25\\" height=\\"25\\" />
                                                </g>
                                                <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                                                  <rect x=\\"4\\" y=\\"11.8\\" width=\\"17\\" height=\\"1.5\\" transform=\\"translate(-5.2 12.5) rotate(-45)\\" />
                                                  <rect x=\\"11.8\\" y=\\"4\\" width=\\"1.5\\" height=\\"16.97\\" transform=\\"translate(-5.2 12.5) rotate(-45)\\" />
                                                </g>
                                              </g>
                                            </svg>
                                          </IconBase>
                                        </Close>
                                      </div>
                                    </button>
                                  </ForwardRef>
                                  <Form labelAlwaysAbove={true} includeSubmitButton={false} requiredVariation=\\"blueBarWithLegend\\" className=\\"fe_u_margin--large document-form\\" layout=\\"medium\\" buttonText=\\"Submit\\" noValidate={true} nested={true} labelWidth={3}>
                                    <form onSubmit={[undefined]} className=\\"fe_u_margin--large document-form\\" noValidate={true}>
                                      <GridRow nested={true}>
                                        <div className=\\"fe_l_grid-row fe_l_grid-row--nested\\">
                                          <GridCol>
                                            <div className=\\"fe_l_grid-row__col\\">
                                              <FormLegend allFieldsRequired={false} nested={true}>
                                                <div className=\\"fe_c_form-legend\\" aria-hidden=\\"true\\">
                                                  Required field
                                                </div>
                                              </FormLegend>
                                            </div>
                                          </GridCol>
                                        </div>
                                      </GridRow>
                                      <FormField maxlength=\\"1000\\" labelText=\\"Link\\" name=\\"filePath\\" value=\\"https://www.drive.in/cover\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} error={[undefined]} id=\\"filePath\\" labelWidth={12} required={true} inputAs={{...}}>
                                        <FormFieldLayout labelWidth={12} nested={[undefined]} layout={[undefined]} statusExtras={[undefined]} className=\\"\\" labelAlwaysAbove={[undefined]} labelSlot={{...}} messageSlot={[undefined]} statusIndicatorSlot={[undefined]} inputSlot={{...}}>
                                          <div className=\\"fe_c_form-field fe_c_form-field--medium\\">
                                            <GridRow nested={true}>
                                              <div className=\\"fe_l_grid-row fe_l_grid-row--nested\\">
                                                <GridCol width={{...}} className=\\"fe_c_form-field__label fe_c_form-field__label--forced-above\\">
                                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-12 fe_c_form-field__label fe_c_form-field__label--forced-above\\">
                                                    <Label id=\\"filePath-label\\" htmlFor=\\"filePath\\" text=\\"Link\\" disabled={[undefined]} required={false} useLegend={false}>
                                                      <label htmlFor=\\"filePath\\" id=\\"filePath-label\\" className=\\"fe_c_label\\">
                                                        Link
                                                      </label>
                                                    </Label>
                                                  </div>
                                                </GridCol>
                                                <GridCol width={{...}} className=\\"fe_c_form-field__right fe_c_form-field__right--label-forced-above\\">
                                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-0 fe_c_form-field__right fe_c_form-field__right--label-forced-above\\">
                                                    <div className=\\"fe_c_form-field__input-slot\\">
                                                      <Input maxlength=\\"1000\\" value=\\"https://www.drive.in/cover\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} className=\\"fe_c_form-field__input\\" id=\\"filePath\\" name=\\"filePath\\" required={true} hideRequiredStyles={false} disabled={false} error={[undefined]} aria-labelledby=\\"filePath-label\\" type=\\"text\\">
                                                        <div className=\\"fe_c_input fe_c_form-field__input\\">
                                                          <input maxlength=\\"1000\\" value=\\"https://www.drive.in/cover\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} id=\\"filePath\\" name=\\"filePath\\" aria-labelledby=\\"filePath-label\\" className=\\"fe_c_input__input fe_is-required\\" type=\\"text\\" disabled={false} required={true} />
                                                        </div>
                                                      </Input>
                                                    </div>
                                                    <div className=\\"fe_c_form-field__status\\" />
                                                  </div>
                                                </GridCol>
                                              </div>
                                            </GridRow>
                                          </div>
                                        </FormFieldLayout>
                                      </FormField>
                                      <FormField maxlength=\\"200\\" labelText=\\"Name\\" name=\\"fileName\\" value=\\"Marksheet\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} error={[undefined]} id=\\"fileName\\" labelWidth={12} required={true} inputAs={{...}}>
                                        <FormFieldLayout labelWidth={12} nested={[undefined]} layout={[undefined]} statusExtras={[undefined]} className=\\"\\" labelAlwaysAbove={[undefined]} labelSlot={{...}} messageSlot={[undefined]} statusIndicatorSlot={[undefined]} inputSlot={{...}}>
                                          <div className=\\"fe_c_form-field fe_c_form-field--medium\\">
                                            <GridRow nested={true}>
                                              <div className=\\"fe_l_grid-row fe_l_grid-row--nested\\">
                                                <GridCol width={{...}} className=\\"fe_c_form-field__label fe_c_form-field__label--forced-above\\">
                                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-12 fe_c_form-field__label fe_c_form-field__label--forced-above\\">
                                                    <Label id=\\"fileName-label\\" htmlFor=\\"fileName\\" text=\\"Name\\" disabled={[undefined]} required={false} useLegend={false}>
                                                      <label htmlFor=\\"fileName\\" id=\\"fileName-label\\" className=\\"fe_c_label\\">
                                                        Name
                                                      </label>
                                                    </Label>
                                                  </div>
                                                </GridCol>
                                                <GridCol width={{...}} className=\\"fe_c_form-field__right fe_c_form-field__right--label-forced-above\\">
                                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-0 fe_c_form-field__right fe_c_form-field__right--label-forced-above\\">
                                                    <div className=\\"fe_c_form-field__input-slot\\">
                                                      <Input maxlength=\\"200\\" value=\\"Marksheet\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} className=\\"fe_c_form-field__input\\" id=\\"fileName\\" name=\\"fileName\\" required={true} hideRequiredStyles={false} disabled={false} error={[undefined]} aria-labelledby=\\"fileName-label\\" type=\\"text\\">
                                                        <div className=\\"fe_c_input fe_c_form-field__input\\">
                                                          <input maxlength=\\"200\\" value=\\"Marksheet\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} id=\\"fileName\\" name=\\"fileName\\" aria-labelledby=\\"fileName-label\\" className=\\"fe_c_input__input fe_is-required\\" type=\\"text\\" disabled={false} required={true} />
                                                        </div>
                                                      </Input>
                                                    </div>
                                                    <div className=\\"fe_c_form-field__status\\" />
                                                  </div>
                                                </GridCol>
                                              </div>
                                            </GridRow>
                                          </div>
                                        </FormFieldLayout>
                                      </FormField>
                                      <FormField labelText=\\"Type\\" inputAs={{...}} options={{...}} name=\\"attachmentTypeId\\" value=\\"1\\" onChange={[Function: onChange]} onBlur={[Function (anonymous)]} error={[undefined]} id=\\"attachmentTypeId\\" labelWidth={12} required={true}>
                                        <FormFieldLayout labelWidth={12} nested={[undefined]} layout={[undefined]} statusExtras={[undefined]} className=\\"\\" labelAlwaysAbove={[undefined]} labelSlot={{...}} messageSlot={[undefined]} statusIndicatorSlot={[undefined]} inputSlot={{...}}>
                                          <div className=\\"fe_c_form-field fe_c_form-field--medium\\">
                                            <GridRow nested={true}>
                                              <div className=\\"fe_l_grid-row fe_l_grid-row--nested\\">
                                                <GridCol width={{...}} className=\\"fe_c_form-field__label fe_c_form-field__label--forced-above\\">
                                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-12 fe_c_form-field__label fe_c_form-field__label--forced-above\\">
                                                    <Label id=\\"attachmentTypeId-label\\" htmlFor=\\"attachmentTypeId\\" text=\\"Type\\" disabled={[undefined]} required={false} useLegend={false}>
                                                      <label htmlFor=\\"attachmentTypeId\\" id=\\"attachmentTypeId-label\\" className=\\"fe_c_label\\">
                                                        Type
                                                      </label>
                                                    </Label>
                                                  </div>
                                                </GridCol>
                                                <GridCol width={{...}} className=\\"fe_c_form-field__right fe_c_form-field__right--label-forced-above\\">
                                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-0 fe_c_form-field__right fe_c_form-field__right--label-forced-above\\">
                                                    <div className=\\"fe_c_form-field__input-slot\\">
                                                      <Select options={{...}} value=\\"1\\" onChange={[Function: onChange]} onBlur={[Function (anonymous)]} className=\\"fe_c_form-field__input\\" id=\\"attachmentTypeId\\" name=\\"attachmentTypeId\\" required={true} hideRequiredStyles={false} disabled={false} error={[undefined]} aria-labelledby=\\"attachmentTypeId-label\\" hasNullDefault={true} placeholder=\\"- Select -\\">
                                                        <select className=\\"fe_c_select fe_is-required fe_c_form-field__input\\" disabled={false} required={true} value=\\"1\\" onChange={[Function: onChange]} onBlur={[Function (anonymous)]} id=\\"attachmentTypeId\\" name=\\"attachmentTypeId\\" aria-labelledby=\\"attachmentTypeId-label\\">
                                                          <option value=\\"\\">
                                                            - Select -
                                                          </option>
                                                          <option value=\\"1\\" disabled={[undefined]} className={[undefined]}>
                                                            Local Rule Authorization
                                                          </option>
                                                          <option value=\\"2\\" disabled={[undefined]} className={[undefined]}>
                                                            Payer Documentation
                                                          </option>
                                                          <option value=\\"3\\" disabled={[undefined]} className={[undefined]}>
                                                            Data Support
                                                          </option>
                                                          <option value=\\"4\\" disabled={[undefined]} className={[undefined]}>
                                                            Athenanet Screenshot
                                                          </option>
                                                          <option value=\\"5\\" disabled={[undefined]} className={[undefined]}>
                                                            Client clarification
                                                          </option>
                                                        </select>
                                                      </Select>
                                                    </div>
                                                    <div className=\\"fe_c_form-field__status\\" />
                                                  </div>
                                                </GridCol>
                                              </div>
                                            </GridRow>
                                          </div>
                                        </FormFieldLayout>
                                      </FormField>
                                      <FormField labelText=\\"Notes\\" name=\\"description\\" value=\\"Test cyril content 1\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} error={[undefined]} id=\\"description\\" labelWidth={12} required={false} maxLength=\\"1000\\" inputAs={{...}}>
                                        <FormFieldLayout labelWidth={12} nested={[undefined]} layout={[undefined]} statusExtras={[undefined]} className=\\"\\" labelAlwaysAbove={[undefined]} labelSlot={{...}} messageSlot={[undefined]} statusIndicatorSlot={[undefined]} inputSlot={{...}}>
                                          <div className=\\"fe_c_form-field fe_c_form-field--medium\\">
                                            <GridRow nested={true}>
                                              <div className=\\"fe_l_grid-row fe_l_grid-row--nested\\">
                                                <GridCol width={{...}} className=\\"fe_c_form-field__label fe_c_form-field__label--forced-above\\">
                                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-12 fe_c_form-field__label fe_c_form-field__label--forced-above\\">
                                                    <Label id=\\"description-label\\" htmlFor=\\"description\\" text=\\"Notes\\" disabled={[undefined]} required={false} useLegend={false}>
                                                      <label htmlFor=\\"description\\" id=\\"description-label\\" className=\\"fe_c_label\\">
                                                        Notes
                                                      </label>
                                                    </Label>
                                                  </div>
                                                </GridCol>
                                                <GridCol width={{...}} className=\\"fe_c_form-field__right fe_c_form-field__right--label-forced-above\\">
                                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-0 fe_c_form-field__right fe_c_form-field__right--label-forced-above\\">
                                                    <div className=\\"fe_c_form-field__input-slot\\">
                                                      <Input value=\\"Test cyril content 1\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} maxLength=\\"1000\\" className=\\"fe_c_form-field__input\\" id=\\"description\\" name=\\"description\\" required={false} hideRequiredStyles={false} disabled={false} error={[undefined]} aria-labelledby=\\"description-label\\" type=\\"text\\">
                                                        <div className=\\"fe_c_input fe_c_form-field__input\\">
                                                          <input value=\\"Test cyril content 1\\" onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} maxLength=\\"1000\\" id=\\"description\\" name=\\"description\\" aria-labelledby=\\"description-label\\" className=\\"fe_c_input__input\\" type=\\"text\\" disabled={false} required={false} />
                                                        </div>
                                                      </Input>
                                                    </div>
                                                    <div className=\\"fe_c_form-field__status\\" />
                                                  </div>
                                                </GridCol>
                                              </div>
                                            </GridRow>
                                          </div>
                                        </FormFieldLayout>
                                      </FormField>
                                    </form>
                                  </Form>
                                  <div className=\\"fe_c_lightbox__footer\\">
                                    <ForwardRef text=\\"Cancel\\" variant=\\"secondary\\" className=\\"fe_u_margin--right-small\\" onClick={[Function: onClick]} disabled={false} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
                                      <button className=\\"fe_c_button fe_c_button--secondary fe_c_button--medium fe_u_margin--right-small\\" aria-disabled={false} aria-label={[undefined]} onClick={[Function: handleClick]} type=\\"button\\">
                                        <div className=\\"fe_c_button__content\\">
                                          <span className=\\"fe_c_button__text fe_c_button__text--medium fe_c_button__text--secondary\\">
                                            Cancel
                                          </span>
                                        </div>
                                      </button>
                                    </ForwardRef>
                                    <ForwardRef text=\\"Save\\" onClick={[Function: onClick]} variant=\\"primary\\" disabled={false} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
                                      <button className=\\"fe_c_button fe_c_button--primary fe_c_button--medium\\" aria-disabled={false} aria-label={[undefined]} onClick={[Function: handleClick]} type=\\"button\\">
                                        <div className=\\"fe_c_button__content\\">
                                          <span className=\\"fe_c_button__text fe_c_button__text--medium fe_c_button__text--primary\\">
                                            Save
                                          </span>
                                        </div>
                                      </button>
                                    </ForwardRef>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </Transition>
                        </CSSTransition>
                      </FadeUp>
                    </Portal>
                  </WaitForContainer>
                </Portal>
              </Modal>
            </Overlay>
          </ModalManagerWrapper>
        </Lightbox>
      </ThemeConsumerHOC>
    </Formik>
  </div>
</AddDocument>"
`;
